### PR TITLE
Expand test coverage to `99%`✨ with edge case tests

### PR DESCRIPTION
**Pull request by qizh**
## Summary

Expand test coverage from `~80%` to **`99%`✨** by adding edge case tests and documenting untestable code paths.

### New Tests
- **`CaseValueMacroTests`**
  Parameter name same as case name, property collision with type/number suffix
- **`CaseNameMacroTests`**
  Enum with non-case members (computed properties, methods)
- **`IsNotCaseMacroTests`**
  Enum with non-case members
- **`OptionSetMacroTests`**
  Custom `optionsName` argument, invalid string literal, Options enum with non-case members
- **`DiagnosticTests`**
  `Diagnostic.note` and `Diagnostic.remark` extension tests

### CaseValueGenerator Enhancement
- Single-case enums now generate **non-optional** properties (avoids "Default will never be executed" warning)

### Documentation
- **`AGENTS.md`**
  Added auto-commit policy with conditions and commit message format
- **`KNOWN_ISSUES.md`**
  Documented untestable `_QizhMacroKitMacro.swift` plugin entry point

### Code Style (`OptionSetGenerator`)
- Refactored to modern Swift style with implicit returns
- ✨ Line **`69`**: Nice. ✨

## Testing
- **`120` tests passing** across `23` suites
- All macro generators covered with edge case tests

<details><summary> <h3>Previously in this PR</h3></summary>

This pull request adds comprehensive documentation and improves platform compatibility handling for the macro implementation code. The most significant change is the addition of the `AGENTS.md` file, which documents project structure, coding conventions, macro implementation patterns, and agent responsibilities. Additionally, the import of `SwiftCompilerPlugin` in `StringifyGenerator.swift` is now conditionally compiled to support non-macOS platforms.

**Documentation and Project Guidelines:**

- Added `AGENTS.md` with detailed information on project overview, structure, coding conventions, macro implementation patterns, agent responsibilities, and CI/CD practices.

**Platform Compatibility:**

- Changed the import of `SwiftCompilerPlugin` in `StringifyGenerator.swift` to be conditional using `#if canImport(SwiftCompilerPlugin)`, improving compatibility with platforms where this module is unavailable.
</details>

### CODE REVIEW

File: Sources/QizhMacroKitMacros/OptionSetGenerator.swift [hunk 1]
```diff
@@ -223,8 +223,8 @@ extension OptionSetGenerator: MemberMacro {
 extension DeclModifierSyntax {
 	var isNeededAccessLevelModifier: Bool {
 		switch self.name.tokenKind {
-		case .keyword(.public): return true
-		default: return false
+		case .keyword(.public): true
+		default: 				false
```
[the remark this answers is not in the document]
Comment (qizh) [reply Sources/QizhMacroKitMacros/OptionSetGenerator.swift:227]: Currently `true` is horizontally aligned with `false`

File: Sources/QizhMacroKitMacros/OptionSetGenerator.swift [hunk 2]
```diff
@@ -29,74 +29,69 @@ extension OptionSetMacroDiagnostic: DiagnosticMessage {
...
-		case .requiresStringLiteral(let name):
-			return "'OptionSet' macro argument \(name) must be a string literal"
-
-		case .requiresOptionsEnum(let name):
-			return "'OptionSet' macro requires nested options enum '\(name)'"
-
-		case .requiresOptionsEnumRawType:
-			return "'OptionSet' macro requires a raw type"
+		case .requiresStruct: 					"'OptionSet' macro can only be applied to a struct"
+		case .requiresStringLiteral(let name): 	"'OptionSet' macro argument '\(name)' must be a string literal"
```
[the remark this answers is not in the document]
Comment (qizh) [reply Sources/QizhMacroKitMacros/OptionSetGenerator.swift:33]: It is intentional and is matching [line 34](https://github.com/qizh/QizhMacroKit/pull/44/changes/BASE..d9360b04a774dde628170213ecd863b7a4857a53#diff-612e95e5461d40354abe75c34117a4e7a641674df3027dc92f6c8c2106855f38R34) quotes.

File: Tests/IsCaseTests/IsCaseMacroTests.swift [hunk 3]
```diff
@@ -236,5 +236,109 @@ struct IsCaseMacroTests {
...
+				/// A parameterless representation of `Status` cases.
+				enum Cases: Equatable, CaseIterable {
+			        case on
+			        case off
+				}
+				/// A parameterless representation of this case.
+				var parametersErasedCase: Cases {
+					switch self {
+			        case .on: .on
+			        case .off: .off
```
[the remark this answers is not in the document]
Comment (qizh) [reply Tests/IsCaseTests/IsCaseMacroTests.swift]: ✦ Nice c⍢tch!
